### PR TITLE
Fix a flaky test that relies on terminate_after for an exact count match

### LINKED ISSUE #10435: [BUG] Search requests with size=0 & terminate_after sometimes return incorrectly hit count
**Describe the bug**
When both `size=0` and `terminate_after` parameters are provided to a search request, the incorrect total hit count is sometimes returned. This was discovered as a part of the investigation for a flakey test, see https://github.com/opensearch-project/OpenSearch/issues/9946#issuecomment-1749266330

**To Reproduce**
Set size=0 in the search request in the below test.
```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.simple.SimpleSearchIT" -Dtests.method="testSimpleTerminateAfterCount {p0={"search.concurrent_segment_search.enabled":"true"}}" -Dtests.seed=C37FD3C5A66D02BC -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=he-IL -Dtests.timezone=Asia/Qostanay -Druntime.java=17
```

**Expected behavior**
The above test should always pass

### PR DESCRIPTION
**Pull request by austintlee**
### Description
[Describe what this change achieves]

### Related Issues
Resolves #10435 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [ x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by austintlee** [comment]
Looks like there were actually two test cases that had @AwaitsFix applied on this issue.  I will take a look at the other one (I only addressed one of them).

File: server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java [hunk 1]
```diff
@@ -305,7 +305,12 @@ public void dotestSimpleTerminateAfterCountWithSize(int size, int max) throws Ex
                 .setSize(size)
                 .setTrackTotalHits(true)
                 .get();
-            assertHitCount(searchResponse, i);
+
+            if (size == 0) {
+                assertHitCount(searchResponse, i, max);
```
Comment (reta) [inline server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java:310]: @austintlee could you please add a reference to Apache Lucene pull request or ticket that explains the behaviour?  thank you

Comment (austintlee) [reply server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java:310]: Will do.

**Comment by austintlee** [comment]
@jed326

**Comment by jed326** [comment]
@austintlee apologies for the super delayed response on this. The changes themselves look good to me but it looks like there are some gradle precommit failures: https://github.com/opensearch-project/OpenSearch/actions/runs/6641276053/job/18043413433?pr=10836

Could you please double check?

**Review by jed326 — APPROVED** [review]

**Comment by dblock** [comment]
@austintlee want to finish this one?

**Comment by austintlee** [comment]
@reta Can you take a look please?

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@austintlee could you please rebase? thank you

**Comment by austintlee** [comment]
@reta my branch is a bit out of date now.  What is the right way to rebase?

**Comment by reta** [comment]
> @reta my branch is a bit out of date now. What is the right way to rebase?

Usually I do: `get fetch origin && git rebase origin/main` for my branches

**Comment by austintlee** [comment]
@reta Can you take another look?  I am not sure why I have DCO failing.

**Comment by austintlee** [comment]
@reta One of the tests is now failing locally with _and_ without concurrent search enabled.  I need to look into that.  (oh boy)

**Comment by reta** [comment]
> @reta One of the tests is now failing locally with _and_ without concurrent search enabled. I need to look into that. (oh boy)

Oh man, @austintlee  is it related to your change? If not, could you share what test you struggle with? Thank you.

**Comment by austintlee** [comment]
`testSimpleTerminateAfterTrackTotalHitsUpToSize`

**Comment by reta** [comment]
> `testSimpleTerminateAfterTrackTotalHitsUpToSize`

It supposed to be fixed, right? Or it looks like we need to re-evaluate the cause?

**Comment by austintlee** [comment]
@reta strangely, I had a change that was not related to the size=0 case.  I reverted it.

Now, my local repo was completely out of sync with upstream so I had to do some git reset and rebase to get my diff to show up correctly and prepared a new PR - https://github.com/opensearch-project/OpenSearch/pull/12179.  Do you think this can still make the 2.12 train?

**Comment by andrross** [comment]
I believe this is superseded by #12179 so I'm closing it. Please correct me if I'm wrong @austintlee

**Pull request closed without merging**